src/DGtal/geometry/doc/moduleDigitalConvexity.dox

```diff
@@ -0,0 +1,165 @@
+i.e. \f$ C^d_k \lbrack Y \rbrack := \{ c \in C^d_k,~\text{s.t.}~ \bar{c} \cap
+Y \neq \emptyset \} \f$.
+
+A digital set \f$ S \subset \mathbb{Z}^d \f$ is said to be \b
+digitally \a k- \b convex whenever \f$ C^d_k \lbrack \mathrm{Conv}(S)
+\rbrack = C^d_k \lbrack S \rbrack \f$. \f$ S \f$ is said to be \b
+fully (digitally) \b convex whenever it is digitally \a k- convex
+for \f$ 0 \le k \le d \f$.
+
+A fully convex set is always \f$ 3^d-1 \f$-connected (i.e. 8-connected
+in 2D, 26-connected in 3D). Furthermore its axis-aligned slices are
+connected (with the same kind of connectedness). It is also clear that
+digitally 0-convexity is the usual digital convexity.
+
+@image html non-fully-convex-triangles.png  "Examples of non fully digitally convex triangles in \f$ \mathbb{Z}^2 \f$. Missing 1-cells for 1-digital convexity are in blue, Missing 2-cells for 2-digital convexity are in green."
+
+@image latex non-fully-convex-triangles.png  "Examples of non fully digitally convex triangles in \f$ \mathbb{Z}^2 \f$. Missing 1-cells for 1-digital convexity are in blue, Missing 2-cells for 2-digital convexity are in green." width=6cm
+
+A last useful notion is the \e subconvexity. Let \f$ X \subset
+\mathbb{Z}^d \f$ some arbitrary digital set. Then the digital set \f$
+S \subset \mathbb{Z}^d \f$ is said to be \b digitally \a k- \b
+subconvex to \f$ X \f$ whenever \f$ C^d_k \lbrack \mathrm{Conv}(S)
+\rbrack \subset C^d_k \lbrack X \rbrack \f$. And \f$ S \f$ is said to
+be \b fully (digitally) \b subconvex to \f$ X \f$ whenever it is
+digitally \a k- subconvex to \f$ X \f$ for \f$ 0 \le k \le d \f$.
+
+Subconvexity is a useful for notion for digital contour and surface
+analysis. It tells which subsets of these digital sets are \e tangent
+to them.
+
+@section dgtal_dconvexity_sec2  Checking digital convexity
+
+Three classes help to check digital convexity.
+
+- BoundedLatticePolytope is the class that is used to build polytopes,
+  i.e. intersections of half-spaces, which are a way to represent
+  convex polyhedra.
+
+- CellGeometry is used to store sets of cells and provides methods to
+  build the set of cells that intersect a polytope or the set of cells
+  that touches a set of digital points.
+
+- DigitalConvexity provides many helper methods to build
+  BoundedLatticePolytope s and CellGeometry s and to check digital
+  convexity and subconvexity.
+
+
+@subsection dgtal_dconvexity_sec21 Building a polytope
+
+For now, the class is quite limited. You may build a polytope from a
+range of \f$ d + 1 \f$ points in general position. The polytope is
+then a simplex. Otherwise you may provide a range of half-spaces. 
+
+You may also cut a polytope by a new halfspace
+(BoundedLatticePolytope::cut), count the number of lattice points
+inside, interior or on the boundary (BoundedLatticePolytope::count,
+BoundedLatticePolytope::countInterior,
+BoundedLatticePolytope::countBoundary) or enumerate them.
+
```

Comment:
Maybe you should give more details about the computational cost of the interior/exterior point counting methods (especially in high dim)